src/OptionSettlementEngine.sol

```diff
@@ -292,10 +292,10 @@ contract OptionSettlementEngine is ERC1155, IOptionSettlementEngine {
 
     /// @inheritdoc IOptionSettlementEngine
     function write(uint256 optionId, uint112 amount, uint256 claimId) public returns (uint256) {
-        (uint160 optionKey, uint96 decodedClaimNum) = decodeTokenId(optionId);
+        (uint160 optionKey, uint96 decodedClaimIndex) = decodeTokenId(optionId);
```

Comment:
We chose claimNum intentionally, what's the reasoning behind changing back to claimIndex?

We are overloading claim index concept again (claimIndex here and claimBucketIndex, OptionLotClaimIndex, etc. elsewhere) and a mismatch in local variable assignment from return variables of the various usages of decodeTokenId().

Recommend we either standardize back to claimNum (or another name we like better), or rename usage everywhere to claimIndex and think about how to simplify/clarify the overloading of claim index concept.